java/arcs/core/data/CreateableStorageKey.kt

```diff
@@ -0,0 +1,14 @@
+package arcs.core.data
+
+import arcs.core.storage.StorageKey
+
+/**
+ * This class represents a storage key in a compiled [Plan] with 'create' fate.
+ */
+class CreateableStorageKey(val nameFromManifest: String) : StorageKey("create") {
```

Comment:
should be a data class so separate instances with the same nameFromManifest within a plan end up receiving the same created key.

Reply:
Not sure I understand, there's already a unit test for this that I added, and since StorageKey already overrides .equals() and .hashCode(), separate instances are already equal via toString()  

Making it a data class wouldn't hurt, but it wouldn't change the equality semantics. I'll make the change before landing.